tests/main/interfaces-log-observe/task.yaml

```
@@ -32,10 +45,43 @@ execute: |
     echo "Then the snap is able to access the system logs"
     log-observe-consumer | MATCH 'ok$'
 
+    echo "And make safe syslog(2) calls"
+
+    # Launch a background thread generating klogs
+    # Otherwise the read commands will hang waiting for kernel output
+    gen_kernel_messages(){
+        while true; do
+            echo "0123456789abcdef" > /dev/kmsg
+            sleep 5
+        done
+    }
+    gen_kernel_messages &
+    pid_gen_kernel_messages=$!
+    #shellcheck disable=SC2064
+    trap "kill -9 $pid_gen_kernel_messages" EXIT
+
+    log-observe-consumer.klogctl SYSLOG_ACTION_READ | MATCH 'SUCCESS'
+    log-observe-consumer.klogctl SYSLOG_ACTION_READ_ALL | MATCH 'SUCCESS'
+    log-observe-consumer.klogctl SYSLOG_ACTION_SIZE_UNREAD | MATCH 'SUCCESS'
+    log-observe-consumer.klogctl SYSLOG_ACTION_SIZE_BUFFER | MATCH 'SUCCESS'
+
     if [ "$(snap debug confinement)" = partial ] ; then
         exit 0
     fi
 
+    echo "But can't make unsafe syslog(2) calls"
+    # Check for EACCES for all these, because they are explicitly denied by seccomp
+    log-observe-consumer.klogctl SYSLOG_ACTION_CLOSE | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_OPEN | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_READ_CLEAR | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_CLEAR | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_CONSOLE_OFF | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_CONSOLE_ON | MATCH 'ERROR permission denied'
+    log-observe-consumer.klogctl SYSLOG_ACTION_CONSOLE_LEVEL | MATCH 'ERROR permission denied'
+    # It is important that the following matches EACCES, not EINVAL, as it
+    # is a stand-in for potential future SYSLOG_ACTIONs, which should be denied by default
+    log-observe-consumer.klogctl SYSLOG_ACTION_BAD | MATCH 'ERROR permission denied'
```

and you should be able to use log.Fatalf() now

```suggestion
    log-observe-consumer.klogctl SYSLOG_ACTION_BAD 2>&1| MATCH 'ERROR permission denied'
```